### PR TITLE
feat(cancun): EIP-6780 SELFDESTRUCT semantics

### LINKED ISSUE #11702: feat(cancun): HIP-868 Support Cancun Self-Destruct Semantics in Smart Contract Services
Support [HIP-868](https://hips.hedera.com/hip/hip-868) - "Support Cancun Self-Destruct Semantics in Smart Contract Services"

The existing EVM `SELFDESTRUCT` opcode is changing it's behavior _only in the case that_ the `SELFDESTRUCT` is executed by a contract that was created _prior_ to the current transaction.  All behavior in the case that the `SELFDESTRUCT` is executed by a contract created in the current transaction stays the same.

Specifically, in this case (contract already existed) the contract will _not_ be deleted (any of it: account, bytecodes, storage) but the sweep of hbar and HTS tokens to a beneficiary account will happen.

Hedera is, of necessity, _non-equivalent_ with Ethereum already with respect to this opcode, because the _sweep_ behavior of HTS tokens is regulated by the Hedera security model - where the beneficiary account needs an _allowance_ for each token - and thus `SELFDESTRUCT` will fail if the allowances are missing.  Where, in Ethereum, `SELFDESTRUCT` can never fail.  _This behavior will be maintained._

* Depends on `ContractsV_0_49Module` from #11698 
* Override `SELFDESTRUCT` with the new Cancun semantics for the case of a pre-existing contract.
* Provide tests the _all_ `SELFDESTRUCT` behavior is as specified in the HIP
* Provide tests that characterize the _existing_ behavior (`V0.38` EVM) for all edge cases w.r.t. allowances, alias unlinking, account destruction, etc. where Hedera is (or should be, or could be) different than Ethereum.

### PR DESCRIPTION
**Pull request by david-bakin-sl**
**Description**:

Post-Cancun SELFDESTRUCT only sweeps if contract was _not_ created in same transaction.

**Related issue(s)**:

Fixes #11702 .

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/docs/design/services/smart-contract-service/cancun-fork-support.md [hunk 1]
```diff
@@ -56,6 +56,8 @@ Opcodes that interact with blobs should not cause the contracts to break, but sh
 It is not a goal to introduce blobs into Hedera. It is also not a goal to restrict design space nor to dictate future
 design directions for support or non-support of blobs.
 
+Cancuns support in mono-services.
```
Comment (lukelee-sl) [inline hedera-node/docs/design/services/smart-contract-service/cancun-fork-support.md:59]: nit: Cancun?

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/opcodes/SelfDestructSuite.java [hunk 2]
```diff
@@ -18,6 +18,7 @@
 
 import static com.hedera.services.bdd.junit.TestTags.SMART_CONTRACT;
```
Comment (lukelee-sl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/opcodes/SelfDestructSuite.java:19]: My guess is that this suite needs to be in either `SequentialSuites` or `ConcurrentSuites`.  Probably `SequentialSuites`?

**Review by lukelee-sl — DISMISSED** [review]
Looks good to me.

File: hedera-node/docs/design/services/smart-contract-service/cancun-fork-support.md (hunk 1 above)
Comment (david-bakin-sl) [reply hedera-node/docs/design/services/smart-contract-service/cancun-fork-support.md:59]: yes.  will fix in next cancun pr

**Review by lukelee-sl — APPROVED** [review]
LGTM

**Review by tinker-michaelj — APPROVED** [review]
LGTM, tyvm @david-bakin-sl !
